src/tests/ov_helpers/ov_snippets_models/src/subgraph_matmul.cpp

```diff
@@ -12,9 +12,76 @@
 namespace ov {
 namespace test {
 namespace snippets {
+namespace {
+constexpr float i8_fq_il = -35.0172004;
+constexpr float i8_fq_ih = 34.7436294;
+constexpr float u8_fq_il = 0;
+constexpr float u8_fq_ih = 0.820726;
+
+std::shared_ptr<ov::Node> build_mm_second_input(const ov::element::Type& precision,
+                                                const ov::PartialShape& shape,
+                                                MatMulType type,
+                                                ov::ParameterVector& params) {
+    std::shared_ptr<ov::Node> result;
+    switch (type) {
+        case MatMulType::FullyConnected:
+            return ov::test::utils::make_constant(precision, shape.to_shape());
+        case MatMulType::MatMul: {
+            auto param = std::make_shared<op::v0::Parameter>(precision, shape);
+            params.push_back(param);
+            return param;
+        }
+        default:
+            OPENVINO_THROW("Unexpected MatMulType is passed in build_mm_second_input");
+    }
+}
+} // namespace
+
+std::ostream &operator<<(std::ostream& os, MatMulType type) {
+    switch (type) {
+        case MatMulType::MatMul:
+            return os << "MatMul";
+        case MatMulType::FullyConnected:
+            return os << "FullyConnected";
+        default:
+            OPENVINO_THROW("Unexpected MatMulType.");
+    }
+}
+
+MatMulFunctionBase::MatMulFunctionBase(const std::vector<PartialShape>& inputShapes,
+                                       MatMulType type,
+                                       const std::vector<ov::element::Type>& precisions)
+    : SnippetsFunctionBase(inputShapes),
+      precisions(precisions),
+      matmul_type(type) {
+    if (!precisions.empty()) {
+        OPENVINO_ASSERT(precisions.size() == 2, "Got invalid number of input element types");
+        const bool is_f32 = ov::snippets::utils::everyone_is(element::f32, precisions[0], precisions[1]);
+        const bool is_int8 = ov::snippets::utils::one_of(precisions[0], element::i8, element::u8) && precisions[1] == element::i8;
+        const bool is_bf16 = ov::snippets::utils::everyone_is(element::bf16, precisions[0], precisions[1]);
+        OPENVINO_ASSERT(is_f32 || is_bf16 || is_int8, "Invalid precisions");
+    }
+}
+
+void MatMulFunctionBase::validate_function(const std::shared_ptr<Model> &f) const {
+    OPENVINO_ASSERT(f != nullptr, "The test requires Model to be defined");
+    const auto shape_size = input_shapes.size();
```

Comment:
Minor:
```suggestion
    const auto count_of_shapes = input_shapes.size();
```

Otherwise, I decided that you want to get `ov::shape_size(...)` 😄

Reply:
You are right, I renamed it to avoid confusion. Thanks!